src/tests/account/test_secp256r1.cairo

```diff
@@ -0,0 +1,141 @@
+
+    let x = u256 {
+        low: xlow.try_into().unwrap(), high: (xhigh_and_parity / 2).try_into().unwrap()
+    };
+    let parity = xhigh_and_parity % 2 == 1;
+
+    assert_eq!(x, private_key);
+    assert_eq!(parity, false, "Parity should be even");
+}
+
+#[test]
+fn test_unpack_big_secp256r1_points() {
+    let (big_point_1, big_point_2) = get_points();
+
+    // Check point 1
+
+    let (expected_x, expected_y) = big_point_1.get_coordinates().unwrap_syscall();
+
+    let (xlow, xhigh_and_parity) = StorePacking::pack(big_point_1);
+    let (x, y) = StorePacking::unpack((xlow, xhigh_and_parity)).get_coordinates().unwrap_syscall();
+
+    assert_eq!(x, expected_x);
+    assert_eq!(y, expected_y);
+
+    // Check point 2
+
+    let (expected_x, _) = big_point_2.get_coordinates().unwrap_syscall();
+
+    let (xlow, xhigh_and_parity) = StorePacking::pack(big_point_2);
+    let (x, _) = StorePacking::unpack((xlow, xhigh_and_parity)).get_coordinates().unwrap_syscall();
+
+    assert_eq!(x, expected_x);
+    assert_eq!(y, expected_y);
+}
+
+#[test]
+fn test_secp256r1_serialization() {
+    let (big_point_1, big_point_2) = get_points();
+
+    let mut serialized_point = array![];
+    let mut expected_serialization = array![];
+
+    // Check point 1
+
+    big_point_1.serialize(ref serialized_point);
+    big_point_1.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+
+    assert!(serialized_point == expected_serialization);
+
+    // Check point 2
+
+    big_point_2.serialize(ref serialized_point);
+    big_point_2.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+
+    assert!(serialized_point == expected_serialization);
+}
+
+#[test]
+fn test_secp256r1_deserialization() {
+    let (big_point_1, big_point_2) = get_points();
+
+    // Check point 1
+
+    let mut expected_serialization = array![];
+
+    big_point_1.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+    let mut expected_serialization = expected_serialization.span();
+    let deserialized_point = Secp256r1PointSerde::deserialize(ref expected_serialization).unwrap();
+
+    assert_eq!(big_point_1, deserialized_point);
+
+    // Check point 2
+
+    let mut expected_serialization = array![];
+
+    big_point_2.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+    let mut expected_serialization = expected_serialization.span();
+    let deserialized_point = Secp256r1PointSerde::deserialize(ref expected_serialization).unwrap();
+
+    assert_eq!(big_point_2, deserialized_point);
+}
+
+#[test]
+fn test_partial_eq() {
+    let (big_point_1, big_point_2) = get_points();
+
+    assert_eq!(big_point_1, big_point_1);
+    assert_eq!(big_point_2, big_point_2);
+    assert!(big_point_1 != big_point_2);
+    assert!(big_point_2 != big_point_1);
+}
+
+//
+// Helpers
+//
+
+/// This signature was computed using @noble/curves.
+fn P256_PRIVATEKEY_SAMPLE() -> u256 {
```

Comment:
Since we are not verifying signatures for this test, Is this function needed?

Reply:
see comment my next comment. I will remove it, once we agree on what to name it.